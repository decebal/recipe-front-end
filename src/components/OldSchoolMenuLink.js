// @flow
/* eslint-disable react/no-children-prop */

import React from 'react';
import PropTypes from 'prop-types';
import { Route, Link } from 'react-router-dom';

const OldSchoolMenuLink = ({ label, to, activeOnlyWhenExact }
: {label: string, to: string, activeOnlyWhenExact: boolean}) => (
  <Route
    path={to}
    exact={activeOnlyWhenExact}
    children={({ match }) => (
      <div className={match ? 'active' : ''}>
        <Link to={to}>{label}</Link>
      </div>
    )}
  />
);

OldSchoolMenuLink.propTypes = {
  label: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  activeOnlyWhenExact: PropTypes.bool,
};

OldSchoolMenuLink.defaultProps = {
  activeOnlyWhenExact: false,
};

export default OldSchoolMenuLink;
