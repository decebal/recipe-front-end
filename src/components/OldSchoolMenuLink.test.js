import React from 'react';
import renderer from 'react-test-renderer';
import OldSchoolMenuLink from './OldSchoolMenuLink';

test('OldSchoolMenuLink renders correctly', () => {
  const tree = renderer
    .create(<OldSchoolMenuLink label="Home" to="/" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
