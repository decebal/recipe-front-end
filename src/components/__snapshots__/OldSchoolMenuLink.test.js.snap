// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OldSchoolMenuLink renders correctly 1`] = `
<div
  className="active"
>
  &gt; 
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;
